claat/proto-renderer/oneof_utils.go

```diff
@@ -0,0 +1,48 @@
+package genrenderer
+
+import (
+  "text/template"
+
+  "github.com/googlecodelabs/tools/third_party"
+)
+
+// RenderOneof returns the underyling, rendered, passed oneof element
+func RenderOneof(el interface{}, t *template.Template) string {
+  // Recursive redering happens here
+  return ExecuteTemplate(AssertOneof(el), t)
+}
+
+// AssertOneof turns a generic oneof proto into its underlying typed-proto
+func AssertOneof(el interface{}) (underlyingType interface{}) {
```

Comment:
I feel like strong typing, like Go provides, should eliminate the need for this. We're basically just saying "what type is this thing?", no?

Reply:
Per the entire reflection discussion on the design doc:
- The only type in Go that can be used as generic/catch-up type is `interface{}`

What's not documented here per assumption of features of golang: The original type of an non-typed interface is lost once we send it down to another method. Therefore, this cannot be eliminated - since when we call `ExecuteTemplate`, `templateName` will fail to get the passed type.

Any suggestions on the function name welcomed. I can only think of `AssertUnderlyingOneof`, since that's the nomenclature I've been using.

Reply:
> should eliminate the need for this
If by `the need for this` do you mean "no need to have an extra function", and the intent is to merge both `RenderOneof` and `AssertOneof` into a single function, this was done for the following:
- Separation of concerns: type-assertions has nothing to with actual template rendering
- Later reusability: if later needed, `AssertOneof` can be used without further refactoring